### PR TITLE
add jest for automatic unit testing.

### PR DESCRIPTION
**Pull request by dev-matan-tsuberi**
Changes:
- added `jest` along with other deps, configuration for `jest` is in `package.json["jest"]`.
- two simple tests to make sure it's all working: [`reducers/notificationsReducer.spec.ts`,`App.spec.tsx`].

Commands:
- `npm run test`
- `npm run test:watch` - rerun relevant tests upon file changes.

Jest also generates coverage reports (in `coverage/`), which can be useful.

### CODE REVIEW

**Comment by dev-matan-tsuberi** [comment]
[Add automated tests](https://trello.com/c/pjdvXnPi/127-add-automated-tests)

**Comment by dev-matan-tsuberi** [comment]
[Add automated tests](https://trello.com/c/pjdvXnPi/127-add-automated-tests)

File: src/App.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import {render} from 'react-dom';
```
Comment (tibetsprague) [inline src/App.spec.tsx]: you think its better to keep specs in the same directory as the component instead of a separate directory?

Comment (tibetsprague) [reply src/App.spec.tsx]: im ok with either

Comment (dev-matan-tsuberi) [reply src/App.spec.tsx]: I slightly prefer keeping them in the same dir, i guess.

**Pull request merged**